### ISSUE TITLE
Make batchnorm gammas trainable for reset

### ISSUE THREAD

**Issue by gonzalezjo** [LeelaChessZero/lczero-training#41]
jio on the Discord suggested this. He never got around to making an issue for it, as far as I saw. I figured I would make one, lest it be forgotten.

> [2:36 PM] Error323: Why weren't they I wonder. Was this not required in leelago @jio ?
> [2:44 PM] jio: @Error323 It's an old oversight in LZ network architecture
> [2:44 PM] jio: they should be trainable in LZ too
> 2:45 PM] jio: however they can't be enabled while maintaining backwards compatibility so it's a little tricky to enable them
> [2:45 PM] Error323: Yeah, so before a new run is an excellent idea. Thanks.

**Comment by Tilps** [comment]
Note that the reason we only turned on batchnorm betas for training and not gammas was because the network file format doesn't have enough variables in it to store gamma information as well as weights and beta.
We could add gammas just to training, but then it wouldn't be possible to start training again from a network file as the gamma values can't be separated back out from the weights and betas.
Alternatively it requires changes to the file format, lc0 and training.

**Comment by gonzalezjo** [comment]
> Alternatively it requires changes to the file format, lc0 and training.


What do these change entail?

**Comment by dubslow** [comment]
It would mean changing the network weights file format, which implies changing all code that handles it -- such as the training code (to write the new data), and the lc0 reading code (to read the new data).

**Comment by Tilps** [comment]
Also potentially the lc0 engine code as well to process the gammas how each engine wants (even if maybe that will just be to back propagate the multiplication into the weights and biases).

**Comment by gonzalezjo** [comment]
This can be closed thanks to the work on SE.
